docs/en/integrations/data-ingestion/data-formats/templates-regex.md

```diff
@@ -0,0 +1,233 @@
+2023/01/16 06:02:09 [error]  client: 8.4.2.7, server: example.com "GET /apple-touch-icon-120x120.png HTTP/1.1"
+2023/01/15 13:46:13 [error]  client: 6.9.3.7, server: example.com "GET /apple-touch-icon.png HTTP/1.1"
+2023/01/16 05:34:55 [error]  client: 9.9.7.6, server: example.com "GET /h5/static/cert/icon_yanzhengma.png HTTP/1.1"
+```
+
+We can use a [Template](https://clickhouse.com/docs/en/interfaces/formats/#format-template) format to import this data. We have to define a template string with values placeholders for each row of input data:
+
+```
+<time> [error] client: <ip>, server: <host> "<request>"
+```
+
+Let's create a table to import our data into:
+```sql
+CREATE TABLE error_log
+(
+    `time` DateTime,
+    `ip` String,
+    `host` String,
+    `request` String
+)
+ENGINE = MergeTree
+ORDER BY (host, request, time)
+```
+
+To import data using a given template, we have to save our template string in a file ([row.template](assets/row.template) in our case):
+```
+${time:Escaped} [error]  client: ${ip:CSV}, server: ${host:CSV} ${request:JSON}
+```
+
+We define a name of a column and escaping rule in a `${name:escaping}` format. Multiple options are available here, like CSV, JSON, Escaped, or Quoted, which implement [respective escaping rules](https://clickhouse.com/docs/en/interfaces/formats/#format-template).
+
+Now we can use the given file as an argument to the `format_template_row` settings option while importing data (*note, that template and data files **should not have** an extra `\n` symbol at the end of file*):
+
+```sql
+INSERT INTO error_log FROM INFILE 'error.log'
+SETTINGS format_template_row = 'row.template'
+FORMAT Template
+```
+
+And we can make sure our data was loaded into the table:
+
+```sql
+SELECT
+    request,
+    count(*)
+FROM error_log
+GROUP BY request;
+
+┌─request──────────────────────────────────────────┬─count()─┐
+│ GET /img/close.png HTTP/1.1                      │     176 │
+│ GET /h5/static/cert/icon_yanzhengma.png HTTP/1.1 │     172 │
+│ GET /phone/images/icon_01.png HTTP/1.1           │     139 │
+│ GET /apple-touch-icon-precomposed.png HTTP/1.1   │     161 │
+│ GET /apple-touch-icon.png HTTP/1.1               │     162 │
+│ GET /apple-touch-icon-120x120.png HTTP/1.1       │     190 │
+└──────────────────────────────────────────────────┴─────────┘
+```
+
+### Skipping whitespaces
+Consider using [TemplateIgnoreSpaces](https://clickhouse.com/docs/en/interfaces/formats/#templateignorespaces), which allows skipping whitespaces between delimiters in a template:
+```
+Template:               -->  "p1: ${p1:CSV}, p2: ${p2:CSV}"
+TemplateIgnoreSpaces    -->  "p1:${p1:CSV}, p2:${p2:CSV}"
+```
+
+## Exporting data using templates
+
+We can also export data to any text format using templates as well. In this case, we have to create two files:
+
+[**Result set template**](), which defines the layout for the whole result set:
```

Comment:
The error says line 80 is missing a link; what markdown file is this supposed to go to?

Reply:
Fixed, and also fixed URLs, sorry, completely missed this.

Reply:
no problem!